### DISCUSSION TITLE
montage now into third hour of processing

### DISCUSSION THREAD

**Discussion by bdrichards** [ImageMagick/ImageMagick#3208] [Help] [answered]
General Question:
I am taking tiled images and recombining them back into a single, large image (have used VIPS, but it has many quirks). Testing montage first with a smaller set of files (~3000, each tile is 256x256 pixels) and found it functional in a short amount of time, several minutes.
Now, however, I am processing 12,276 tiles (198 columns by 62 rows) and the process seems to be running still after 3 hours (lots of 1Mb files in the TEMP folder still being made, over 15,000 so far)!!! Is this to be expected?
My system: Win10 SurfaceBook with 
Installed Physical Memory (RAM)	8.00 GB
Total Physical Memory	7.92 GB
Available Physical Memory	1.52 GB
Total Virtual Memory	11.7 GB
Available Virtual Memory	1.99 GB
Page File Space	3.80 GB

... and the command line: magick montage *.jpg -geometry +0+0 -tile 198x62 all.jpg
Should I cut my losses, stop the process, and move to a 128GB RAM  media laptop with faster CPU?

**Comment by snibgo** [comment, marked as answer]
From the temp files, we know your pixels are being cached on disk. That is why it takes so long.

12276 images of 256x256 pixels is 805 million pixels. If you are using HDRI IM, this will break your 8.00GB memory, even before any processing. As your inputs are JPG, you could try an IM build that is Q8 integer.

You could montage in batches, then append the result together.

If you have a machine with 128GB memory, that seems a good idea.

**Reply by bdrichards** [reply]
Ok, thanks!
I will have to wait a few days, then, that other laptop is in a quarantined office! I will definitely install the Q8 version there, thanks for that suggestion. What if the input files were PNG, would I loose some pixel bit-depth?

**Reply by bdrichards** [reply]
As another question:
In order to test montage on batches, say the first 5 rows:
magick montage *.jpg -geometry +0+0 -tile 198x5 all.jpg

still appears to process all 12,000+ images. The files are named sequentially with 5-digit names starting with "00000.jpg"  up to "12275.jpg".  I can create a dozen or so temp directories and copy over images in rows of five or so, but How do you specify to only process the first 990 images (which is 198*5)??

**Comment by snibgo** [comment]
> What if the input files were PNG, would I loose some pixel bit-depth?

That would make no difference.

My comment about JPEG is that those files are 8-bit/channel integer files, so you wouldn't care that the outputs are also 8-bit/channel integer. The memory taken by an image depends on the IM build, and the number of pixels and channels per pixel. It does not depend on the input file format.

> How do you specify to only process the first 990 images (which is 198*5)??

Write the filenames of the first 990 images to a text file, say called "myfiles.txt". Then call magick montage with "@myfiles.txt" as the input filename. Then repeat with the next 990 files, and so on.

**Reply by bdrichards** [reply]
Ok, thank you!
As for processing a specific group to "batch" the process, I thought some type of format specifier like this might work for the input range of files, :
>magick  %d.jpg[00000-00990] ......

but no success unless there is a format statement that would work.

Creating the input txt file with file names is also a possibility , but I think I will settle on just creating  a series of subfolders with copies of a group of 20 rows at a time (198 columns x 20 rows operates pretty good even on the 8GB machine). Issue is that we have undergrad research students to process hundreds of the large tilesets and we need simplest process with little room for error.
Thank you for your clarifications!
